packages/fxa-shared/metrics/amplitude-event.1.schema.json

```diff
@@ -197,7 +235,53 @@
         }
       },
       "then": {
-        "required": ["language"]
+        "required": ["language"],
+        "properties": {
+          "user_properties": {
+            "required": ["flow_id"]
+          },
+          "event_properties": {
+            "required": ["plan_id", "product_id"]
+          }
+        }
+      }
+    },
+    {
+      "if": {
+        "$comment": "For event_type's matching the pattern below, make following fields required",
+        "required": ["event_type"],
+        "properties": {
+          "event_type": {
+            "type": "string",
+            "pattern": "^fxa_pay_setup -"
+          }
+        }
+      },
+      "then": {
+        "properties": {
+          "event_properties": {
+            "required": ["checkout_type", "subscribed_plan_ids"]
```

Comment:
Per your Redux logic (and my related comment), `subscribed_plan_ids` is going to be optional now.

Reply:
👍